hack/dev-env/start-principal.sh

```
@@ -44,13 +44,20 @@ E2E_ENV_FILE="/tmp/argocd-agent-e2e"
 if [ -f "$E2E_ENV_FILE" ]; then
     source "$E2E_ENV_FILE"
     export ARGOCD_PRINCIPAL_ENABLE_WEBSOCKET=${ARGOCD_PRINCIPAL_ENABLE_WEBSOCKET:-false}
+    export ARGOCD_PRINCIPAL_ENABLE_SELF_CLUSTER_REGISTRATION=${ARGOCD_PRINCIPAL_ENABLE_SELF_CLUSTER_REGISTRATION:-false}
+    if [ -n "$ARGOCD_PRINCIPAL_RESOURCE_PROXY_TLS_CA_PATH" ]; then
+        export ARGOCD_PRINCIPAL_RESOURCE_PROXY_TLS_CA_PATH
+    fi
 fi
 
+export ARGOCD_AGENT_RESOURCE_PROXY=$(ip r show default | sed -e 's,.*\ src\ ,,' | sed -e 's,\ metric.*$,,')
+
 SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
```

Shellcheck (0.11.0)

[warning] SCRIPTPATH appears unused. Verify use (or export if used externally).

(SC2034)